### PR TITLE
Test new cartesian controller

### PR DESCRIPTION
**Pull request by fmessmer**
Code-Review PR for commenting on @ipa-fxm-cm's changes
Supersedes #9 

@ipa-fxm-cm please merge with my `test_new_cartesian_controller` branch (i.e. `git pull origin-fxm test_new_cartesian_controller`) before applying more changes to your test_new_cartesian_controller branch as I merged with latest ipa320/indigo_dev (only minor changes, but it's cleaner like that!)

### CODE REVIEW

File: cob_cartesian_controller/include/cob_cartesian_controller/cartesian_controller_data_types.h [hunk 1]
```diff
@@ -55,8 +62,6 @@ namespace cob_cartesian_controller
         double start_angle, end_angle;
         double radius;
         bool rotate_only;
```
Comment (fmessmer) [inline cob_cartesian_controller/include/cob_cartesian_controller/cartesian_controller_data_types.h]: can `rotate_only` be removed here, too?

File: cob_cartesian_controller/include/cob_cartesian_controller/trajectory_profile_generator/trajectory_profile_generator_base.h [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+ *   and Automation (IPA) \n\n
+ *
+ *****************************************************************
+ *
+ * \note
+ *   Project name: care-o-bot
+ * \note
+ *   ROS stack name: cob_control
+ * \note
+ *   ROS package name: cob_twist_controller
```
Comment (fmessmer) [inline cob_cartesian_controller/include/cob_cartesian_controller/trajectory_profile_generator/trajectory_profile_generator_base.h]: cob_cartesian_controller

File: cob_cartesian_controller/include/cob_cartesian_controller/trajectory_profile_generator/trajectory_profile_generator_base.h [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+ *
+ * \author
+ *   Author: Christoph Mark, email: christoph.mark@ipa.fraunhofer.de / christoph.mark@gmail.com
+ *
+ * \date Date of creation: June, 2015
+ *
+ * \brief
+ *   This header contains the interface description of all available
+ *   hardware interface types (position/velocity).
```
Comment (fmessmer) [inline cob_cartesian_controller/include/cob_cartesian_controller/trajectory_profile_generator/trajectory_profile_generator_base.h]: please update copy-pasted `brief` and `date`

File: cob_cartesian_controller/include/cob_cartesian_controller/trajectory_profile_generator/trajectory_profile_generator_base.h [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+ *
+ ****************************************************************/
+#ifndef HARDWARE_INTERFACE_TYPE_BASE_H_
+#define HARDWARE_INTERFACE_TYPE_BASE_H_
```
Comment (fmessmer) [inline cob_cartesian_controller/include/cob_cartesian_controller/trajectory_profile_generator/trajectory_profile_generator_base.h]: please update header guard!

File: cob_cartesian_controller/include/cob_cartesian_controller/trajectory_profile_generator/trajectory_profile_generator_base.h [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+
+        virtual bool calculateProfile(std::vector<double>* path_matrix,
+                                      const double Se_lin, const double Se_rot,
+                                      geometry_msgs::Pose start) = 0;
+
+        virtual std::vector<double> getTrajectory(double se, double accl, double vel, double t_ipo,
+                                                  double steps_tb, double steps_tv, double steps_te, double tb, double tv, double te) = 0;
+    private:
+        virtual bool generatePath(cob_cartesian_controller::PathArray &pa) = 0;
+//        virtual cob_cartesian_controller::ProfileTimings getProfileTimings(double te_max, double accl, double vel) = 0;
```
Comment (fmessmer) [inline cob_cartesian_controller/include/cob_cartesian_controller/trajectory_profile_generator/trajectory_profile_generator_base.h]: can this line be removed?

File: cob_cartesian_controller/include/cob_cartesian_controller/trajectory_profile_generator/trajectory_profile_generator_base.h [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+
+
+    protected:
+        const cob_cartesian_controller::CartesianActionStruct &params_;
+        double vel_max_;
+};
+
+#endif /* HARDWARE_INTERFACE_TYPE_BASE_H_ */
```
Comment (fmessmer) [inline cob_cartesian_controller/include/cob_cartesian_controller/trajectory_profile_generator/trajectory_profile_generator_base.h]: update

File: cob_cartesian_controller/include/cob_cartesian_controller/trajectory_profile_generator/trajectory_profile_generator_ramp.h [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+ *   ROS stack name: cob_control
+ * \note
+ *   ROS package name: cob_cartesian_controller
+ *
+ * \author
+ *   Author: Christoph Mark, email: christoph.mark@ipa.fraunhofer.de / christoph.mark@gmail.com
+ *
+ * \date Date of creation: September, 2015
+ *
+ * \brief
```
Comment (fmessmer) [inline cob_cartesian_controller/include/cob_cartesian_controller/trajectory_profile_generator/trajectory_profile_generator_ramp.h:24]: please write a short `brief` description

File: cob_cartesian_controller/include/cob_cartesian_controller/trajectory_profile_generator/trajectory_profile_generator_builder.h [hunk 8]
```diff
@@ -0,0 +1,45 @@
...
+ *   ROS stack name: cob_control
+ * \note
+ *   ROS package name: cob_cartesian_controller
+ *
+ * \author
+ *   Author: Christoph Mark, email: christoph.mark@ipa.fraunhofer.de / christoph.mark@gmail.com
+ *
+ * \date Date of creation: September, 2015
+ *
+ * \brief
```
Comment (fmessmer) [inline cob_cartesian_controller/include/cob_cartesian_controller/trajectory_profile_generator/trajectory_profile_generator_builder.h:24]: please write a short `brief` description

File: cob_cartesian_controller/include/cob_cartesian_controller/trajectory_profile_generator/trajectory_profile_generator_ramp.h [hunk 9]
```diff
@@ -0,0 +1,56 @@
...
+ *
+ ****************************************************************/
+#ifndef COB_CONTROL_COB_CARTESIAN_CONTROLLER_INCLUDE_COB_CARTESIAN_CONTROLLER_TRAJECTORY_PROFILE_GENERATOR_TRAJECTORY_PROFILE_GENERATOR_RAMP_H_
+#define COB_CONTROL_COB_CARTESIAN_CONTROLLER_INCLUDE_COB_CARTESIAN_CONTROLLER_TRAJECTORY_PROFILE_GENERATOR_TRAJECTORY_PROFILE_GENERATOR_RAMP_H_
+
+
+#include "cob_cartesian_controller/trajectory_profile_generator/trajectory_profile_generator_base.h"
+
+/* BEGIN TrajectoryProfileRamp ****************************************************************************************/
+/// Class providing a HardwareInterface publishing velocities.
```
Comment (fmessmer) [inline cob_cartesian_controller/include/cob_cartesian_controller/trajectory_profile_generator/trajectory_profile_generator_ramp.h]: please update!

File: cob_cartesian_controller/include/cob_cartesian_controller/trajectory_profile_generator/trajectory_profile_generator_sinoid.h [hunk 10]
```diff
@@ -0,0 +1,56 @@
...
+ *   ROS stack name: cob_control
+ * \note
+ *   ROS package name: cob_cartesian_controller
+ *
+ * \author
+ *   Author: Christoph Mark, email: christoph.mark@ipa.fraunhofer.de / christoph.mark@gmail.com
+ *
+ * \date Date of creation: September, 2015
+ *
+ * \brief
```
Comment (fmessmer) [inline cob_cartesian_controller/include/cob_cartesian_controller/trajectory_profile_generator/trajectory_profile_generator_sinoid.h:24]: please write a short `brief` description

File: cob_cartesian_controller/include/cob_cartesian_controller/trajectory_profile_generator/trajectory_profile_generator_sinoid.h [hunk 11]
```diff
@@ -0,0 +1,56 @@
...
+ *
+ ****************************************************************/
+#ifndef COB_CONTROL_COB_CARTESIAN_CONTROLLER_INCLUDE_COB_CARTESIAN_CONTROLLER_TRAJECTORY_PROFILE_GENERATOR_TRAJECTORY_PROFILE_GENERATOR_SINOID_H_
+#define COB_CONTROL_COB_CARTESIAN_CONTROLLER_INCLUDE_COB_CARTESIAN_CONTROLLER_TRAJECTORY_PROFILE_GENERATOR_TRAJECTORY_PROFILE_GENERATOR_SINOID_H_
+
+#include "cob_cartesian_controller/trajectory_profile_generator/trajectory_profile_generator_base.h"
+
+/* BEGIN TrajectoryProfileSinoid ****************************************************************************************/
+/// Class providing a HardwareInterface publishing positions.
```
Comment (fmessmer) [inline cob_cartesian_controller/include/cob_cartesian_controller/trajectory_profile_generator/trajectory_profile_generator_sinoid.h]: please update!

File: cob_cartesian_controller/include/cob_cartesian_controller/trajectory_profile_generator/trajectory_profile_generator_sinoid.h [hunk 12]
```diff
@@ -0,0 +1,56 @@
...
+        virtual cob_cartesian_controller::ProfileTimings getProfileTimings(double Se, double te, double accl, double vel, bool calcMaxTe);
+        virtual bool generatePath(cob_cartesian_controller::PathArray &pa);
+        virtual bool calculateProfile(std::vector<double>* path_matrix,
+                                      const double Se_lin, const double Se_rot,
+                                      geometry_msgs::Pose start);
+        virtual std::vector<double> getTrajectory(double se, double accl, double vel, double t_ipo,
+                                                  double steps_tb, double steps_tv, double steps_te, double tb, double tv, double te);
+
+    private:
+//        virtual cob_cartesian_controller::ProfileTimings getProfileTimings(double Se, double te, double accl, double vel);
```
Comment (fmessmer) [inline cob_cartesian_controller/include/cob_cartesian_controller/trajectory_profile_generator/trajectory_profile_generator_sinoid.h]: can this line be removed?

File: cob_cartesian_controller/msg/MoveCirc.msg [hunk 13]
```diff
@@ -6,5 +6,3 @@ float64 end_angle
 float64 radius
 
 bool rotate_only
```
Comment (fmessmer) [inline cob_cartesian_controller/msg/MoveCirc.msg]: can we remove rotate_only?

File: cob_cartesian_controller/msg/MoveLin.msg [hunk 14]
```diff
@@ -2,5 +2,3 @@ geometry_msgs/Pose pose_goal
 string frame_id
 
 bool rotate_only
```
Comment (fmessmer) [inline cob_cartesian_controller/msg/MoveLin.msg]: can we remove rotate_only?

File: cob_cartesian_controller/src/cartesian_controller.cpp [hunk 15]
```diff
@@ -197,12 +197,12 @@ bool CartesianController::movePTP(const geometry_msgs::Pose& target_pose, const
 // Broadcasting interpolated Cartesian path
 bool CartesianController::posePathBroadcaster(const geometry_msgs::PoseArray& cartesian_path)
 {
-    bool success = false;
+    bool success = true;
```
Comment (fmessmer) [inline cob_cartesian_controller/src/cartesian_controller.cpp:155]: `success` should be assumed `false` at the beginning

File: cob_cartesian_controller/src/cartesian_controller.cpp [hunk 16]
```diff
@@ -212,36 +212,42 @@ bool CartesianController::posePathBroadcaster(const geometry_msgs::PoseArray& ca
...
         // Get transformation
         tf::StampedTransform stamped_transform = utils_.getStampedTransform(target_frame_, chain_tip_link_);
 
-        // Check whether chain_tip_link is within epsilon area of target_frame
-        if(!utils_.inEpsilonArea(stamped_transform, epsilon))
+        ros::spinOnce();
+        rate.sleep();
+    }
+
+    while(true)
```
Comment (fmessmer) [inline cob_cartesian_controller/src/cartesian_controller.cpp]: should not be `while(true)`...
maybe we can use `while (ros::ok())` or something action-related, `as_->` (http://docs.ros.org/indigo/api/actionlib/html/classactionlib_1_1SimpleActionServer.html)

File: cob_cartesian_controller/src/cartesian_controller.cpp [hunk 17]
```diff
@@ -212,36 +212,42 @@ bool CartesianController::posePathBroadcaster(const geometry_msgs::PoseArray& ca
...
+    {
+        // Get transformation
+        tf::StampedTransform stamped_transform = utils_.getStampedTransform(target_frame_, chain_tip_link_);
+        ROS_INFO_STREAM("stamped: x: " << stamped_transform.getOrigin().getX() << " y: " << stamped_transform.getOrigin().getY() << " z: " << stamped_transform.getOrigin().getZ());
```
Comment (fmessmer) [inline cob_cartesian_controller/src/cartesian_controller.cpp]: I guess once verified, this ROS_INFO output can be removed

File: cob_cartesian_controller/src/cartesian_controller.cpp [hunk 18]
```diff
@@ -212,36 +212,42 @@ bool CartesianController::posePathBroadcaster(const geometry_msgs::PoseArray& ca
...
+
+        if(!utils_.inEpsilonArea(stamped_transform, 0.0005))
```
Comment (fmessmer) [inline cob_cartesian_controller/src/cartesian_controller.cpp]: please use a `#defiine` macro for this value (put it in the header!)

File: cob_cartesian_controller/src/cartesian_controller.cpp [hunk 19]
```diff
@@ -255,10 +261,16 @@ bool CartesianController::posePathBroadcaster(const geometry_msgs::PoseArray& ca
...
 void CartesianController::goalCallback()
 {
+    geometry_msgs::Pose start_pose, end_pose, pose;
     geometry_msgs::PoseArray cartesian_path;
     cob_cartesian_controller::CartesianActionStruct action_struct;
 
+
+    ROS_INFO_STREAM("=========================================================================");
     ROS_INFO("Received a new goal");
+    ROS_INFO_STREAM("=========================================================================");
```
Comment (fmessmer) [inline cob_cartesian_controller/src/cartesian_controller.cpp]: debug...could be removed

File: cob_cartesian_controller/src/cartesian_controller.cpp [hunk 20]
```diff
@@ -290,10 +302,12 @@ void CartesianController::goalCallback()
             return;
         }
 
+//        movePTP(action_struct.move_lin.end, 0.001);
+
```
Comment (fmessmer) [inline cob_cartesian_controller/src/cartesian_controller.cpp]: remove

File: cob_cartesian_controller/src/cartesian_controller.cpp [hunk 21]
```diff
@@ -355,16 +361,16 @@ cob_cartesian_controller::MoveLinStruct CartesianController::convertMoveLin(cons
...
-    move_lin_struct.start = start;
-    move_lin_struct.end = end;
+    move_lin.start = start;
+    move_lin.end = end;
 
-    return move_lin_struct;
+    ROS_WARN("move_lin_msg_orientation .. w: %f", move_lin_msg.pose_goal.orientation.w);
+    ROS_WARN("move_lin_msg_orientation .. x: %f", move_lin_msg.pose_goal.orientation.x);
+    ROS_WARN("move_lin_msg_orientation .. y: %f", move_lin_msg.pose_goal.orientation.y);
+    ROS_WARN("move_lin_msg_orientation .. z: %f", move_lin_msg.pose_goal.orientation.z);
```
Comment (fmessmer) [inline cob_cartesian_controller/src/cartesian_controller.cpp]: debug...could be removed

File: cob_cartesian_controller/src/cartesian_controller_utils.cpp [hunk 22]
```diff
@@ -149,12 +147,44 @@ void CartesianControllerUtils::previewPath(const geometry_msgs::PoseArray& pose_
...
 
-    for(unsigned int i=0; i<pose_array.poses.size(); i++)
+    double id = marker_array_.markers.size();
+
+    for(unsigned int i=0; i < pose_array.poses.size(); i++)
+        {
+            marker.id = id+i;
+            marker.pose = pose_array.poses.at(i);
+            marker_array_.markers.push_back(marker);
+        }
```
Comment (fmessmer) [inline cob_cartesian_controller/src/cartesian_controller_utils.cpp]: indentation

File: cob_cartesian_controller/src/cartesian_controller_utils.cpp [hunk 23]
```diff
@@ -149,12 +147,44 @@ void CartesianControllerUtils::previewPath(const geometry_msgs::PoseArray& pose_
...
-        marker.pose = pose_array.poses.at(i);
-        marker_array.markers.push_back(marker);
+        path_array[i] = m[i].array_;
     }
+}
 
-    marker_pub_.publish(marker_array);
+double CartesianControllerUtils::roundUpToMultiplier(double numberToRound, double multiplier)
+{
+    return ( multiplier - fmod(numberToRound,multiplier) + numberToRound );
```
Comment (fmessmer) [inline cob_cartesian_controller/src/cartesian_controller_utils.cpp]: explicit namespace please: `std::fmod`

File: cob_cartesian_controller/src/trajectory_interpolator/trajectory_interpolator.cpp [hunk 24]
```diff
@@ -27,136 +27,140 @@
...
-    pitch_path   = path_matrix[2];
-    yaw_path     = path_matrix[3];
+    angular_path = path_matrix[1];
+
+    if(linear_path.back() == 0)
+    {
+        norm_factor = 1/angular_path.back();
+        path = angular_path;
+    }
+    else    //todo: There's still a bug in the rotational interpolation !
```
Comment (fmessmer) [inline cob_cartesian_controller/src/trajectory_interpolator/trajectory_interpolator.cpp]: ToDo :wink:

File: cob_cartesian_controller/src/trajectory_interpolator/trajectory_interpolator.cpp [hunk 25]
```diff
@@ -27,136 +27,140 @@
...
+     tf::Transform C, P, T;
+
+     std::vector<double> path_array;
+     std::vector<double> path_matrix[4];
+
+     geometry_msgs::Pose pose;
+
+     double Se = as.move_circ.end_angle - as.move_circ.start_angle;
+
+     bool forward;
```
Comment (fmessmer) [inline cob_cartesian_controller/src/trajectory_interpolator/trajectory_interpolator.cpp:115]: What does "forward" mean? clockwise vs. counter-clockwise?
Does it have something to do with shortest angle?

File: cob_cartesian_controller/src/trajectory_profile_generator/trajectory_profile_generator_builder.cpp [hunk 26]
```diff
@@ -0,0 +1,55 @@
...
+ * \note
+ *   ROS package name: cob_twist_controller
+ *
+ * \author
+ *   Author: Christoph Mark, email: christoph.mark@ipa.fraunhofer.de / christoph.mark@gmail.com
+ *
+ * \date Date of creation: June, 2015
+ *
+ * \brief
+ *   This module contains the implementation of all interface types.
```
Comment (fmessmer) [inline cob_cartesian_controller/src/trajectory_profile_generator/trajectory_profile_generator_builder.cpp]: please update copy-pasted license

File: cob_cartesian_controller/src/trajectory_profile_generator/trajectory_profile_generator_builder.cpp [hunk 27]
```diff
@@ -0,0 +1,55 @@
...
+ *
+ ****************************************************************/
+#define RAMP_PROFILE 1
+#define SINOID_PROFILE 2
```
Comment (fmessmer) [inline cob_cartesian_controller/src/trajectory_profile_generator/trajectory_profile_generator_builder.cpp]: please remove the define makros and use the defines from the messages directly https://github.com/ipa-fxm/cob_control/blob/test_new_cartesian_controller/cob_cartesian_controller/msg/Profile.msg#L1-L2

File: cob_cartesian_controller/src/trajectory_profile_generator/trajectory_profile_generator_builder.cpp [hunk 28]
```diff
@@ -0,0 +1,55 @@
...
+#include "cob_cartesian_controller/trajectory_profile_generator/trajectory_profile_generator_sinoid.h"
+
+/* BEGIN TrajectoryProfileBase *****************************************************************************************/
+
+TrajectoryProfileBase* TrajectoryProfileBuilder::createProfile(const cob_cartesian_controller::CartesianActionStruct& params)
+{
+    TrajectoryProfileBase* ib = NULL;
+    switch(params.profile.profile_type)
+    {
+        case RAMP_PROFILE:
```
Comment (fmessmer) [inline cob_cartesian_controller/src/trajectory_profile_generator/trajectory_profile_generator_builder.cpp]: see comment above

File: cob_cartesian_controller/src/trajectory_profile_generator/trajectory_profile_generator_builder.cpp [hunk 29]
```diff
@@ -0,0 +1,55 @@
...
+            ib = new TrajectoryProfileRamp(params);
+            break;
+        case SINOID_PROFILE:
```
Comment (fmessmer) [inline cob_cartesian_controller/src/trajectory_profile_generator/trajectory_profile_generator_builder.cpp]: see comment above

File: cob_cartesian_controller/src/trajectory_profile_generator/trajectory_profile_generator_ramp.cpp [hunk 30]
```diff
@@ -0,0 +1,155 @@
...
+ *   ROS stack name: cob_control
+ * \note
+ *   ROS package name: cob_cartesian_controller
+ *
+ * \author
+ *   Author: Christoph Mark, email: christoph.mark@ipa.fraunhofer.de / christoph.mark@gmail.com
+ *
+ * \date Date of creation: September, 2015
+ *
+ * \brief
```
Comment (fmessmer) [inline cob_cartesian_controller/src/trajectory_profile_generator/trajectory_profile_generator_ramp.cpp:24]: please add a short `brief` description

File: cob_cartesian_controller/src/trajectory_profile_generator/trajectory_profile_generator_ramp.cpp [hunk 31]
```diff
@@ -0,0 +1,155 @@
...
+    int steps_te, steps_tv, steps_tb = 0;
+    double tv, tb = 0.0;
+
+    // Calculate the Sinoid Profile Parameters
+    if (vel > sqrt(std::fabs(Se) * accl))
+    {
+        vel = sqrt(std::fabs(Se) * accl);
+    }
+
+    if(vel > 0.00001)
```
Comment (fmessmer) [inline cob_cartesian_controller/src/trajectory_profile_generator/trajectory_profile_generator_ramp.cpp]: please use a `#define`
is `0.00001` used as some kind of https://github.com/ipa-fxm/cob_control/blob/test_new_cartesian_controller/cob_twist_controller/include/cob_twist_controller/cob_twist_controller_data_types.h#L45-L46?

File: cob_cartesian_controller/src/trajectory_profile_generator/trajectory_profile_generator_ramp.cpp [hunk 32]
```diff
@@ -0,0 +1,155 @@
...
+    }
+
+    pa.array_ = array;
+    return true;
+}
+
+inline std::vector<double> TrajectoryProfileRamp::getTrajectory(double se, double accl, double vel, double t_ipo, double steps_tb, double steps_tv, double steps_te, double tb, double tv, double te)
+{
+    std::vector<double> array;
+    unsigned int i = 1;
```
Comment (fmessmer) [inline cob_cartesian_controller/src/trajectory_profile_generator/trajectory_profile_generator_ramp.cpp:75]: why not start at `0`? This should broadcast the current pose, right?

File: cob_cartesian_controller/src/trajectory_profile_generator/trajectory_profile_generator_ramp.cpp [hunk 33]
```diff
@@ -0,0 +1,155 @@
...
+    cob_cartesian_controller::PathArray lin(Se_lin, linear_path);
+    cob_cartesian_controller::PathArray rot(Se_rot, angular_path);
+
+
+    cob_cartesian_controller::PathMatrix pm(lin,rot);
+
+    // Get the profile timings from the longest path
+    pt_max_ = getProfileTimings(pm.getMaxSe(), 0, params_.profile.accl, params_.profile.vel, true);
+
+    ROS_INFO_STREAM("pt_max_SEEEEEEEEEEEEE: " << pm.getMaxSe());
```
Comment (fmessmer) [inline cob_cartesian_controller/src/trajectory_profile_generator/trajectory_profile_generator_ramp.cpp]: debug...could be removed

File: cob_cartesian_controller/src/trajectory_profile_generator/trajectory_profile_generator_sinoid.cpp [hunk 34]
```diff
@@ -0,0 +1,155 @@
...
+ *   ROS stack name: cob_control
+ * \note
+ *   ROS package name: cob_cartesian_controller
+ *
+ * \author
+ *   Author: Christoph Mark, email: christoph.mark@ipa.fraunhofer.de / christoph.mark@gmail.com
+ *
+ * \date Date of creation: September, 2015
+ *
+ * \brief
```
Comment (fmessmer) [inline cob_cartesian_controller/src/trajectory_profile_generator/trajectory_profile_generator_sinoid.cpp:24]: please add a short `brief` description

File: cob_cartesian_controller/src/trajectory_profile_generator/trajectory_profile_generator_sinoid.cpp [hunk 35]
```diff
@@ -0,0 +1,155 @@
...
+    int steps_te, steps_tv, steps_tb = 0;
+    double tv, tb = 0.0;
+
+    // Calculate the Sinoid Profile Parameters
+    if (vel > sqrt(std::fabs(Se) * accl / 2))
+    {
+        vel = sqrt(std::fabs(Se) * accl / 2);
+    }
+
+    if(vel > 0.001)
```
Comment (fmessmer) [inline cob_cartesian_controller/src/trajectory_profile_generator/trajectory_profile_generator_sinoid.cpp]: again, use a `#define`

**Comment by fmessmer** [comment]
I only reviewed this PR in the browser!

Beside the comments above, @ipa-fxm-cm still has few issues/ToDo's/ideas himself about optimizing the new cartesian_controller. Please go on finalizing...
Also, please review the code yourself in order to identify lines of code or functions or variables which are no longer needed.....or also debug output (ROS_*) that is no longer needed!

Once finalized and above comments are addressed, @ipa-fxm-cm will open PR from his updated test_new_cartesian_controller branch to ipa320/indigo_dev directly!

File: cob_cartesian_controller/scripts/lwa4d/lwa4d_circle.py [hunk 36]
```diff
@@ -0,0 +1,66 @@
+#! /usr/bin/env python
```
Comment (fmessmer) [inline cob_cartesian_controller/scripts/lwa4d/lwa4d_circle.py:1]: in the end, I think we should remove the lwa4d_\* scripts from cob_cartesian_controller and put them into `schunk_lwa4d` directly!

File: cob_cartesian_controller/scripts/lwa4d/lwa4d_quad.py [hunk 37]
```diff
@@ -0,0 +1,139 @@
+#! /usr/bin/env python
```
Comment (fmessmer) [inline cob_cartesian_controller/scripts/lwa4d/lwa4d_quad.py:1]: in the end, I think we should remove the lwa4d_\* scripts from cob_cartesian_controller and put them into `schunk_lwa4d` directly!

File: cob_cartesian_controller/scripts/lwa4d/lwa4d_quad_angular.py [hunk 38]
```diff
@@ -0,0 +1,121 @@
+#! /usr/bin/env python
```
Comment (fmessmer) [inline cob_cartesian_controller/scripts/lwa4d/lwa4d_quad_angular.py:1]: in the end, I think we should remove the lwa4d_\* scripts from cob_cartesian_controller and put them into `schunk_lwa4d` directly!

File: cob_cartesian_controller/src/trajectory_profile_generator/trajectory_profile_generator_ramp.cpp (hunk 32 above)
Comment (MaChristoph) [inline cob_cartesian_controller/src/trajectory_profile_generator/trajectory_profile_generator_ramp.cpp:75]: This function interpolates the path for time > 0 (here in discrete time i > 0) ... i = 0 would represent the actual position of the tracking_frame which can be neglected.

**Pull request closed without merging**